### DISCUSSION TITLE
Organization-aware login behavior regarding username/email

### DISCUSSION THREAD

**Discussion by gabriel-hoff-vectorsolutions** [zitadel/zitadel#7342] [Q&A]
I have a use case for username/email login that requires usernames & email addresses to be unique at the organization level.

Regarding organization-level unique email addresses, Zitadel doesn't seem to have any feature to enforce that. (Hopefully, I am wrong about this, and you can correct me)

As for organization-level usernames, I can enable "add organization domain as suffix to loginnames". This kind of solves the problem, but it creates another problem: Users now have to be aware of their organization domain and remember to append it to the end of their usernames.

I considered sending the organization's primary domain as a scope parameter (i.e., `urn:zitadel:iam:org:domain:primary:{domainname}`) with the "hide Loginname suffix" option enabled, but doing this prevents users from logging in with their email address.

**Questions:**

- Is this an intended feature?
  - If yes, what is the reason for not allowing domain-enforced logins to use email addresses?
  - If not, have you considered allowing email addresses for domain-enforced logins?

I also considered using the organization's ID as a scope parameter (i.e., `urn:zitadel:iam:org:id:{id}`) since users would be able to log in using email addresses. Still, we would be back at the problem of users having to know and enter their organization's domain.

**Questions:**

- Have you considered the feature of accepting usernames without the domain suffix during organization-enforced logins?
  - If yes, what would be the reason for it?
  - If not, would you be willing to consider it?

I understand that requiring the domain suffix for organization-enforced logins shouldn't be necessary, as usernames without the suffix should always be unique at the organization level.

I am looking forward to getting those answers. Thanks in advance for all the help you have provided so far.

**Comment by hifabienne** [comment]
Hi @vs-gtadeu 
We already have an open issue about the email login when scope is sent: https://github.com/zitadel/zitadel/issues/6868

@livio-a @muhlemmer can you answer the second question?

**Reply by gabriel-hoff-vectorsolutions** [reply]
Hi @hifabienne.

I'm happy to know that it is a known bug. If that bug gets fixed, it will unblock us from using Zitadel for our username/email address login use case. Please let me know as soon as you have any updates on it.

**Reply by gabriel-hoff-vectorsolutions** [reply]
The answer to the second question is optional, now knowing that the issue with domain-enforced logins is a bug. However, an answer would still be appreciated (and might help someone else with a similar question).
